csrc/host_ir/lower_to_communication.cpp

```diff
@@ -346,16 +353,20 @@ std::optional<CommunicationInfo> getCommunicationInfo(Expr* expr) {
         CommunicationType type = same_mesh ? CommunicationType::Allgather
                                            : CommunicationType::Gather;
         create_communication_info(type, p_logical_id, p2c_map.at(p_logical_id));
-        continue;
       }
       if (!p_sharded && c_sharded) {
-        // Scatter
         IterDomain* c_logical_id = getLogicalFromLoopId(consumer, c_loop_did);
         create_communication_info(
             CommunicationType::Scatter, c2p_map.at(c_logical_id), c_logical_id);
-        continue;
       }
-    } else if (expr->isA<ReductionOp>()) {
+      if (p_sharded && c_sharded) {
```

Comment:
This code will always create `SendRecv` for consistently sharded tvs though.
That does not seem right.
If sharded axes are mapped, and meshes are same, this is not a sharding change. This condition appears worth adding for multiple parallel types.

Reply:
It does not have to be in this PR since we are not testing 2D sharding yet.
But sooner than later would be good. My primary intention would be to keep this code as functional for 2D sharding as we can, so we do not need refactors for that later.

Reply:
I created https://github.com/NVIDIA/Fuser/issues/4604 to track this. 

> we do not need refactors for that later

Are you sure it takes only refactors? convertSingleOpToCommunication generates wrong results for 2D sharding, regardless of this PR. Even if this PR didn't change getCommunicationInfo, something like the following
```
in: iDIDx{m}, iDIDy{n}
out = set(in): iDIDx{m}, i{n}
```
would break convertSingleOpToCommunication.

Reply:
> is that just because how we create comms there?

because it uses the same buggy (for 2D sharding) logic as in getCommunicationInfo to decide SendRecv. See #4604.

Reply:
Got it, then resolving the bug in `getCommunicationInfo` will help!